packages/features/ee/sso/lib/sso.ts

```diff
@@ -0,0 +1,57 @@
+import type { PrismaClient } from "@calcom/prisma";
+import { TRPCError } from "@calcom/trpc/server";
+
+import jackson from "./jackson";
+import { tenantPrefix, samlProductID } from "./saml";
+
+export const ssoTenantProduct = async (prisma: PrismaClient, email: string) => {
+  const { connectionController } = await jackson();
+
+  const memberships = await prisma.membership.findMany({
+    select: {
+      teamId: true,
+    },
+    where: {
+      accepted: true,
+      user: {
+        email,
+      },
+    },
+  });
+
```

Comment:
SAML/OIDC connections are stored against the team ID. That means the app needs to determine the team ID the user belongs to to fetch the proper connection and redirect them to the IdP.